mmpose/models/detectors/top_down.py

```diff
@@ -185,9 +185,16 @@ def forward_test(self, img, img_metas, return_heatmap=False, **kwargs):
         # compute backbone features
         output = self.backbone(img)
 
-        return self.process_head(
+        all_preds, all_boxes, image_path, heatmap = self.process_head(
             output, img, img_metas, return_heatmap=return_heatmap)
 
+        # convert the dtype of heatmap to np.float16 and only keep value > 3e-3
+        # (to save space)
+        if heatmap is not None:
+            heatmap = compact_heatmaps(heatmap[0])
```

Comment:
What if the batchsize > 1?

Reply:
We do not support batch size > 1 now because:
At test time, the batch size is 1 (hardcoded in `tools/test.py`). The code will be executed if and only if we use `tools/test.py` and set return_heatmap as True.
I will add an assertion line here to make the condition explicit.

Reply:
Not really. In v0.10.0, we have already supported multi-batch inference (batch_size>1).

https://github.com/open-mmlab/mmpose/blob/3fe69444d1cced87bfaffaeb20da328177d10148/configs/top_down/resnet/coco/res50_coco_256x192.py#L122

Reply:
OK, I will update the code to support multi-batch testing